Assets/Scripts/Boat/BoatController.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BoatController : MonoBehaviour
+{
+    
+    private BoatMovement _boatMovement;
+    public Transform currentDock;
+    
+    public Transform leftDock;
+    public Transform rightDock;
+
+    public KeyCode voyageStartKey;
+ 
+
+    // Start is called before the first frame update
+    void Awake()
+    {
+        _boatMovement = GetComponent<BoatMovement>();
+    }
+
+    private void Start()
+    {
+        currentDock = leftDock;
```

Comment:
Do you want to subscribe to the BoatCapacity's OnBoatDestroyed Event?

Reply:
Slight realization.
OnBoatDestroyed could be replaced with `GameState.Instance.CurrentState = GameStates.End;`

That will call `GameStates.OnEndEnter` which can be subscribed to instead.